### ISSUE TITLE
bad ?lang= in GET request should return an error

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#20452]
The API is pretty lax about `lang` values, and will try to return a matching localization for an addon (etc) if available.  In api/v5+ we return the requested `lang` as `null` if it's not found - if the lang value is garbage data (defined containing other than alphanumeric characters + `_`, `-`) we should return a 400 instead of returning a normal response (in api/v5+ because breaking change).

**Comment by ioanarusiczki** [comment]
@eviljeff 

I keep getting 400 "The request cannot be fulfilled due to bad syntax". When I try requests from -dev or -stage using the addon detail API (I don't even use the ?lang parameter)

https://addons-dev.allizom.org/api/v5/addons/addon/ghosttext@bfred.it/ or https://addons.allizom.org/api/v5/addons/addon/tabliss/  -> I tried with or without authentication , with addon id, slug or guid 

I tried with public versions on -prod and I get 200 responses ,  the problem is not present here
https://addons.mozilla.org/api/v5/addons/addon/foresty-green/

**Comment by eviljeff** [comment]
> @eviljeff
> 
> I keep getting 400 "The request cannot be fulfilled due to bad syntax". When I try requests from -dev or -stage using the addon detail API (I don't even use the ?lang parameter)
> 
> https://addons-dev.allizom.org/api/v5/addons/addon/ghosttext@bfred.it/ or https://addons.allizom.org/api/v5/addons/addon/tabliss/ -> I tried with or without authentication , with addon id, slug or guid

Those links wfm; and with and `lang` param I get the expected 400 error.  This patch isn't on stage yet so if you're getting an error there also it's something unrelated.

**Comment by ioanarusiczki** [comment]
@eviljeff 

I also get the 400 using special chars ->    "The language code \"=\" is invalid."

When I try the detail API from Postman cloud , all responses on -dev and -stage are a 400. -> all I know is that Postman is using a proxy (which could cause this problem). Also from cloud, other APIs are responding without errors , such as the new submission API.  I assume something changed for the details because last week I've tested and this issue was not present. 

I installed the desktop app and tried from here and the responses are 200 on the detail API.
